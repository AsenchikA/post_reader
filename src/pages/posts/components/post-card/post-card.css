.post-card {
  border-radius: 5px;
  border: 1px solid blue;

  &__date {
    padding: 10px;
    border-bottom: 1px solid blue;
  }

  &__message {
    padding: 10px;
  }
}
