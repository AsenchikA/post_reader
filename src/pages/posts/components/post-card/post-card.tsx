import classNames from 'classnames';
import React, { FunctionComponent } from 'react';
import { IShortPostModel } from '../../../../types';
import styles from './post-card.css';

interface IPostCardProps extends IShortPostModel {
  className?: string;
}

export const PostCard: FunctionComponent<IPostCardProps> = ({ className, created_time, message }) => {
  const postDate = new Date(created_time);

  return (
    <div className={classNames(styles['post-card'], className)}>
      <div className={styles['post-card__date']}>{postDate.toUTCString()}</div>
      <div className={styles['post-card__message']}>{message}</div>
    </div>
  );
};
