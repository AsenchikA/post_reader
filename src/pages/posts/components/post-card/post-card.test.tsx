import React from 'react';
import renderer from 'react-test-renderer';
import { PostCard } from './post-card';

it('PostCard renders correcly', () => {
  const props = {
    created_time: '1970-01-01T00:00:00Z',
    message: 'message',
    className: 'test-class',
  };

  const tree = renderer.create(<PostCard {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
