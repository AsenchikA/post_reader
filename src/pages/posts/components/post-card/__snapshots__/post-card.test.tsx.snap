// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCard renders correcly 1`] = `
<div
  className="post-card test-class"
>
  <div
    className="post-card__date"
  >
    Thu, 01 Jan 1970 00:00:00 GMT
  </div>
  <div
    className="post-card__message"
  >
    message
  </div>
</div>
`;
